design-proposals/emts-scale-provisioning.md

```diff
@@ -0,0 +1,142 @@
+  participant pa as Provisioning Nginx
+  participant inv as Inventory / API
+  end
+
+  user->>inv: Pre-register ENs with SN/UUID
+  note over bios,ipxe: PXE boot starts
+
+  bios->>smee: DHCP Discover
+  smee->>bios: DHCP reply with TFTP endpoint storing base EFI script
+
+  bios->>ipxe: Leaves PXE context, taken over by iPXE
+  ipxe->>smee: DHCP Discover
+  smee->>ipxe: DHCP reply with HTTP endpoint where boot.ipxe is stored
+
+  ipxe->>+pa: Download boot.ipxe
+  pa->>-ipxe: [boot.ipxe]
+
+  note over bios,inv: OS provisioning continues in a standard flow, through HookOS and Tinkerbell workflow
+```
+
+Notes:
+- Similar to the standard flow - If EN is not pre-registered, the process will fallback to Interactive Onboarding waiting for a user to provide credentials.
+- Similar to the standard flow - If EN is pre-registered, but the OS profile is not selected, the default OS will be provisioned.
+- In Step 3, SMEE returns an URL to EFI script that is stored on SMEE'S TFTP server. This operation is fully handled by SMEE and it supports various of hardware PXE architectures:
+  https://github.com/tinkerbell/smee/blob/main/internal/dhcp/dhcp.go#L44. 
+- In Step 6, SMEE replies with HTTP(S) URL to iPXE script stored on Provisioning Nginx. We need custom SMEE configuration (mentioned above) to make it happen.
+- The rest of steps after Step 6 follows the existing OS provisioning flow. In summary, the difference between HTTPS-based boot and PXE boot is how the iPXE script is triggered.
+
+### Rationale
+
+The current design proposal allows to easily support legacy PXE boot while keeping the current UX around EN pre-registration. It also lets us to keep all the features that EIM currently supports,
+including logging, KPI instrumentation, observability, etc.
+
+The alternative considered was to use a standalone Tinkerbell deployment without the rest of the EIM stack. While it has clear advantage of a more lightweight deployment,
+it would completely change the UX as we would need to familiarize customers with Tinkerbell APIs (or have a custom CLI tool to help them manage Tinkerbell CRDs).
+With the current proposal we keep using the current UX, with possibility to use Bulk Import Tool to scale preregistration process and selectively choose OS profiles for Edge Nodes.
+
+## Affected components and Teams
+
+- **FPS team** should work with the help of EIM team on slimming down the FPS components to meet EIM-S requirements.
+- **EIM team** responsible for slimming down deployment of EIM and enabling Tinkerbell SMEE with required configuration. 
+
+## Open issues
+
+- By default, the provisioning flow will complete and all the agents will be installed, started and connected to the orchestrator instance that deployed them.
+  We may need a way to stop the provisioning flow without installing and starting agents.
+- In this solution, PXE boot will always be allowed for any device in a local subnet trying to initiate PXE boot.
```

Comment:
I'm ok with this assumption, we assume the OXM takes security of his network and also that it has a know list of devices trying to PXE boot